src/Containers/MassTransit.AutofacIntegration/Registration/ContainerBuilderRegistrationConfigurator.cs

```diff
@@ -155,15 +136,20 @@ static ISendEndpointProvider GetCurrentSendEndpointProvider(IComponentContext co
             if (context.TryResolve(out ConsumeContext consumeContext))
                 return consumeContext;
 
-            return context.Resolve<IBus>();
+            return new PayloadSendEndpointProvider<ILifetimeScope>(context.Resolve<IBus>(), GetPayloadFactory(context));
         }
 
         static IPublishEndpoint GetCurrentPublishEndpoint(IComponentContext context)
         {
             if (context.TryResolve(out ConsumeContext consumeContext))
                 return consumeContext;
 
-            return context.Resolve<IBus>();
+            return new PublishEndpoint(new PayloadPublishEndpointProvider<ILifetimeScope>(context.Resolve<IBus>(), GetPayloadFactory(context)));
+        }
+
+        static PayloadFactory<ILifetimeScope> GetPayloadFactory(IComponentContext context)
+        {
+            return context.Resolve<Func<ILifetimeScope>>().Invoke;
```

Comment:
I don't think the Func<ILifetimeScope> is necessary, since it's going to be created immediately anyway. Just get it and return it.